x-pack/plugins/apm/public/context/apm_service/apm_service_context.tsx

```diff
@@ -97,7 +97,7 @@ export function getOrRedirectToTransactionType({
   transactionType?: string;
   transactionTypes: string[];
   agentName?: string;
-  history: History;
+  history?: History;
```

Comment:
I don't think we should make this optional, because it's easy to forget. If you need only the current transaction type, you'll probably want to split up the function (e.g. you'd create `getTransactionType` that doesn't execute a redirect and use that).

Reply:
I agree. Let's rename it to `getTransactionType` and inline the `replace` call in `ApmServiceContextProvider`:

```ts
  const transactionType = getTransactionType({
    transactionType: query.transactionType,
    transactionTypes,
    agentName,
  });

  // Replace transactionType in the URL in case it is not one of the types returned by the API
  replace(history, { query: { transactionType } });
```

Reply:
done